### PR TITLE
[ Fix ] Improve error message for hooks check

### LINKED ISSUE #4356: Improve error message for hooks check
`instead got object` is weird - better to check for `null` and print that? or `util.inspect` perhaps

_Originally posted by @SimenB in https://github.com/fastify/fastify/pull/4351#discussion_r997819390_

### PR DESCRIPTION
**Pull request by debadutta98**
This PR Closes #4356 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/hooks.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ Hooks.prototype.validate = function (hook, fn) {
   if (supportedHooks.indexOf(hook) === -1) {
     throw new Error(`${hook} hook not supported!`)
   }
-  if (typeof fn !== 'function') throw new FST_ERR_HOOK_INVALID_HANDLER(hook, typeof fn)
+  if (typeof fn !== 'function') throw new FST_ERR_HOOK_INVALID_HANDLER(hook, fn === null ? null : typeof fn)
```
Comment (jsumners) [inline lib/hooks.js:55]: It'd be far easier to read, and have the desired result, to use `Object.prototype.toString.call(fn)` instead.

Comment (debadutta98) [reply lib/hooks.js]: Thanks @jsumners ,
I think now I implemented this in the way you want it to be

File: lib/hooks.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ Hooks.prototype.validate = function (hook, fn) {
   if (supportedHooks.indexOf(hook) === -1) {
     throw new Error(`${hook} hook not supported!`)
   }
-  if (typeof fn !== 'function') throw new FST_ERR_HOOK_INVALID_HANDLER(hook, typeof fn)
+  if (typeof fn !== 'function') throw new FST_ERR_HOOK_INVALID_HANDLER(hook, fn === null ? Object.prototype.toString.call(fn) : typeof fn)
```
[the remark this answers is not in the document]
Comment (debadutta98) [reply lib/hooks.js]: Ok my bad now it's fine

image

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]

**Pull request merged**